libs/common/ring_buffer.hpp

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_COMMON_RING_BUFFER_HPP
+#define IROHA_COMMON_RING_BUFFER_HPP
+
+namespace iroha { namespace containers {
+  /**
+   * Ring buffer implementation with static array layout.
+   * Can be used on stack.
+   * @tparam T - type of element of the buffer
+   * @tparam Count - number of the elements in the buffer
+   */
+    template<typename T, size_t Count> class RingBuffer final {
+    public:
+        using Type = T;
+        using Handle = size_t;
+
+    private:
+        static_assert(Count > 0, "Unexpected count value. It must be above 0.");
+
+        enum { kSizeType = sizeof(Type) };
```

Comment:
```suggestion
        enum { kTypeSize = sizeof(Type) };
```
Because "size type" sounds more like `size_t`, which is the type, but not the "size of the type".